cranelift/reader/src/parser.rs

```diff
@@ -921,6 +921,49 @@ impl<'a> Parser<'a> {
         }
     }
 
+    // Match and consume an i128 immediate.
+    fn match_imm128(&mut self, err_msg: &str) -> ParseResult<i128> {
```

Comment:
I see the duplication is pre-existing, but it's also unfortunate to continue it here. While it may expand the scope of this PR a bit, I wonder if there's a way to write a single polymorphic implementation? Maybe it can take the signed/unsigned types and require `FromStr` / `FromHex` / whatever encapsulates `wrapping_neg`?

Reply:
Perfect, thank you!